### PR TITLE
Exchange rate cloud function

### LINKED ISSUE #420: Move exchange writing to Cloud Function
Move the exchange rate querying from the notification service to a scheduled Cloud Function.

See https://www.notion.so/clabsco/Notification-Service-Indexer-064ca4f58be64c3384ac8fb5f00fca56 for more information on the motivation for doing this.

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Moved the exchange rate querying from the notification service to a scheduled Cloud Function.

See https://www.notion.so/clabsco/Notification-Service-Indexer-064ca4f58be64c3384ac8fb5f00fca56 for more information on the motivation for doing this.

### Other changes

N/A

### Tested

Deployed in Alfajores, checked that the exchange rates are still written every 30 minutes.

### How others should test

Make sure that the exchange rate graph on Valora reflects the right information.

### Related issues

- Fixes #420

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/cloud-functions/src/exchangeRate/index.ts [hunk 1]
```diff
@@ -54,22 +60,39 @@ export async function handleExchangeQuery() {
...
+  const exchangeRateRecord: ExchangeRateObject = {
+    exchangeRate,
+    timestamp,
+  }
+  database().ref(`/exchangeRates/${pair}`).push(exchangeRateRecord)
+  console.debug(`Recorded exchange rate for ${pair}`, exchangeRateRecord)
+}
+
+export const updateExchangeRates = functions.pubsub
+  .schedule('*/30 * * * *')
```
Comment (jeanregisser) [inline packages/cloud-functions/src/exchangeRate/index.ts:97]: nit: 
```suggestion
  .schedule('*/30 * * * *') // every 30 mins
```

**Review by jeanregisser — APPROVED** [review]
Great! 👍 

In the long run we'd want to even remove this exchange rate write task and update the consumers so they query this directly from `blockscout`.
@carterqw2 was telling me `blockscout` stores this internally (at least for cUSD / CELO) for the price graph displayed on the web ui.

**Comment by gnardini** [comment]
> In the long run we'd want to even remove this exchange rate write task and update the consumers so they query this directly from blockscout.

That's a possibility. If we want more independence from blockscout there's a TODO in the code that says to use some `MedianUpdated` events emitted by the exchange contract, we can consider using that too.

For now though I just wanted to move this out of the notification-service :)

**Comment by Celoqa** [comment]
Hey @gnardini @MuckT I have verified this issue using both Mainnet build V1.15.0  & Alfajores Build V1.14.3(62) for iOS and V1.14.1 for Android & observed following: 
**Observations:** 

- **For Mainnet:**
- [x] User is able to see same exchange rate in  two different account addresses. [Attachment](https://drive.google.com/file/d/1swRxY7JdB9We5IToRKSvy_wwJX1CL8S2/view?usp=sharing)
- **For Alfajores:** 
- [ ] User not able to see same exchange rate for two different account addresses. Both rates are different.[Attachment](https://drive.google.com/file/d/1832ZaU3nVa1ZKsMjMxarr54c3k0TZudx/view?usp=sharing)

**FYI:** We have tested above two scenario for different users/account addresses

Can you please help us for how we check that the this Valora exchange rate is showing right information?
Is anything (website or link) where we can compare rate & make sure that it is showing right information or not ?

**Verified On Devices:** Redmi Note 8 (9.0), iPhone SE (14.2)
Thanks..!!

**Comment by MuckT** [comment]
https://valora.testquality.com requirement [R25](https://valora.testquality.com/project/12128/testcase/286107/requirement/11358). Do not remove or edit.

**Pull request merged**